include/seqan/seq_io/bam_sam.h

```diff
@@ -0,0 +1,521 @@
+                        TSeqString const & seq,
+                        TQualString const & qual,
+                        Sam const & tag)
+{
+    _writeRecord(file, meta, seq, qual, tag);
+}
+
+// ----------------------------------------------------------------------------
+// Function writeRecord(SAM); Qualities inside seq
+// ----------------------------------------------------------------------------
+template <typename TSpec, typename TIdString, typename TSeqString>
+inline SEQAN_FUNC_ENABLE_IF(HasQualities<typename Value<TSeqString>::Type> , void)
+writeRecord(FormattedFile<Fastq, Output, TSpec> & file,
+            TIdString const & meta,
+            TSeqString const & seq,
+            Sam const & tag)
+{
+    typedef QualityExtractor<typename Value<TSeqString>::Type> TQualityExtractor;
+    ModifiedString<TSeqString const, ModView<TQualityExtractor> > quals(seq);
+    CharString qual = quals;
+    writeRecord(file, meta, seq, quals, tag);
+}
+
+// ----------------------------------------------------------------------------
+// Function writeRecord(SAM); Without qualities inside seq
+// ----------------------------------------------------------------------------
+template <typename TSpec, typename TIdString, typename TSeqString>
+inline SEQAN_FUNC_ENABLE_IF(Not<HasQualities<typename Value<TSeqString>::Type> >, void)
+writeRecord(FormattedFile<Fastq, Output, TSpec> & file,
+                        TIdString const & meta,
+                        TSeqString const & seq,
+                        Sam const & tag)
+{
+        writeRecord(file, meta, seq, "*", tag);
+}
+// ----------------------------------------------------------------------------
+// Function writeRecord(BAM); Separate Qualities
+// ----------------------------------------------------------------------------
+template <typename TSpec, typename TIdString, typename TSeqString, typename TQualString>
+inline void writeRecord(FormattedFile<Fastq, Output, TSpec> & file,
+                        TIdString const & meta,
+                        TSeqString const & seq,
+                        TQualString const & qual,
+                        Bam const & tag)
+{
+    _writeRecord(file, meta, seq, qual, tag);
+}
+
+// ----------------------------------------------------------------------------
+// Function writeRecord(BAM); Qualities inside seq
+// ----------------------------------------------------------------------------
+template <typename TSpec, typename TIdString, typename TSeqString>
+inline SEQAN_FUNC_ENABLE_IF(HasQualities<typename Value<TSeqString>::Type> , void)
+writeRecord(FormattedFile<Fastq, Output, TSpec> & file,
+            TIdString const & meta,
+            TSeqString const & seq,
+            Bam const & tag)
+{
+    typedef QualityExtractor<typename Value<TSeqString>::Type> TQualityExtractor;
+    ModifiedString<TSeqString const, ModView<TQualityExtractor> > quals(seq);
+    CharString qual = quals;
+    writeRecord(file, meta, seq, quals, tag);
```

Comment:
why do you have to convert to CharString, can you not pass the ModString directly?